[FSDP/AC] checkpoint_wrapper acccept auto_wrap_policy

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #102890
* #102692
* #102680
* __->__ #102672
* #102658
* #102657
* #102639
* #102637
* #102508

Some feedback for this API is that folks would like to use
auto_wrap_policy similar to FSDP instead of having to adapt to the signature of
``check_fn``.

Differential Revision: [D46340320](https://our.internmc.facebook.com/intern/diff/D46340320/)

rohan-varma committed:
[FSDP/AC] checkpoint_wrapper acccept auto_wrap_policy

Some feedback for this API is that folks would like to use
auto_wrap_policy similar to FSDP instead of having to adapt to the signature of
``check_fn``.

Differential Revision: [D46340320](https://our.internmc.facebook.com/intern/diff/D46340320/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[FSDP/AC] checkpoint_wrapper acccept auto_wrap_policy"


Some feedback for this API is that folks would like to use
auto_wrap_policy similar to FSDP instead of having to adapt to the signature of
``check_fn``.

Differential Revision: [D46340320](https://our.internmc.facebook.com/intern/diff/D46340320/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[FSDP/AC] checkpoint_wrapper acccept auto_wrap_policy"


Some feedback for this API is that folks would like to use
auto_wrap_policy similar to FSDP instead of having to adapt to the signature of
``check_fn``.

Differential Revision: [D46340320](https://our.internmc.facebook.com/intern/diff/D46340320/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[FSDP/AC] checkpoint_wrapper acccept auto_wrap_policy"


Some feedback for this API is that folks would like to use
auto_wrap_policy similar to FSDP instead of having to adapt to the signature of
``check_fn``.

Differential Revision: [D46340320](https://our.internmc.facebook.com/intern/diff/D46340320/)

[ghstack-poisoned]

awgu reviewed (approved):
This change makes sense to me, but please check lint and unit test failures.

rohan-varma committed:
Update on "[FSDP/AC] checkpoint_wrapper acccept auto_wrap_policy"


Some feedback for this API is that folks would like to use
auto_wrap_policy similar to FSDP instead of having to adapt to the signature of
``check_fn``.

Differential Revision: [D46340320](https://our.internmc.facebook.com/intern/diff/D46340320/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[FSDP/AC] checkpoint_wrapper acccept auto_wrap_policy"


Some feedback for this API is that folks would like to use
auto_wrap_policy similar to FSDP instead of having to adapt to the signature of
``check_fn``.

Differential Revision: [D46340320](https://our.internmc.facebook.com/intern/diff/D46340320/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[FSDP/AC] checkpoint_wrapper acccept auto_wrap_policy"


Some feedback for this API is that folks would like to use
auto_wrap_policy similar to FSDP instead of having to adapt to the signature of
``check_fn``.

Differential Revision: [D46340320](https://our.internmc.facebook.com/intern/diff/D46340320/)

[ghstack-poisoned]
